commit abd678739ef1bd66bd0bb0d636cf41d8262fb829 (refs/changes/31/431/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-09-07T16:32:50-07:00 (3 years, 1 month ago)
    
    joyent/node-cueball#16 new backends being added causes pool to throw out connections

diff --git a/lib/pool.js b/lib/pool.js
index b40e967..22ad766 100644
--- a/lib/pool.js
+++ b/lib/pool.js
@@ -641,8 +641,8 @@ mod_util.inherits(CueBallConnectionPool, FSM);
 
 CueBallConnectionPool.prototype.on_resolver_added = function (k, backend) {
 	backend.key = k;
-	this.p_keys.push(k);
-	mod_utils.shuffle(this.p_keys);
+	var idx = Math.floor(Math.random() * this.p_keys.length);
+	this.p_keys.splice(idx, 0, k);
 	this.p_backends[k] = backend;
 	this.rebalance();
 };
